### ISSUE TITLE
Writing Julia Objects to pickle and load from python

### ISSUE THREAD

**Issue by chengchingwen** [chengchingwen/Pickle.jl#5]
To make Julia Object loadable from python, there are three issues:
1.  For simple Julia struct (no functions): utilities the stack machine to build a similar python class and load data into it. Will need a rule-base translation for the struct definition to python class definition string.

2. For Julia Arrays: use the array element type to determine whether to write as python list or numpy array.

3. For complex Julia Object (like Function and struct with functions): currently there is no simple solution, so these will be unsupported.

**Comment by findmyway** [comment]
> For Julia Arrays: use the array element type to determine whether to write as python list or numpy array.

Could it be more specific here? I guess you mean `Array{<:Any}` -> `list`, `Array{<:Number}` -> numpy array?

**Comment by chengchingwen** [comment]
Yes, I think we can use some rule to determine whether the `Array` is compatible with numpy array, and if not, fallback to python list.

**Comment by chengchingwen** [comment]
numpy array also have python object dtype, we probably can use it for `Array{Any, N}`, but I haven't try it.
